### PR TITLE
Add FTLD Module functionality to redcap2nacc

### PR DESCRIPTION
**Pull request by s-emerson**
This feature adds the FTLD module to nacculator so that redcap2nacc can process Frontotemporal Lobar Degeneration data from REDCap. An ftld folder has been added to the nacc directory on the same level as uds3 and lbd, and includes ivp and fvp folders with builder and form files. The ftld folder also includes its own blanks file and packet file.
The feature also adds an ftld_blanks_test.py unit test that checks to make sure the custom blanking rule special cases are being picked up by the check_blanks function. Through setting up this unit test I discovered that redcap2nacc will return every blanking rule for a field when it finds an error in the "special cases" list, rather than only the blanking rule that was violated. I can work on fixing that in a new feature branch.
I have updated the README file to include directions for how to use the -lbd and -ftld flags, and I have run the flake8 delinter to clean up reasonable warnings (I did not worry about the forms or builder files). 
This feature addresses issue [#48](https://github.com/ctsit/nacculator/issues/48) .

I also cleaned up the LBD module files a little bit, such as adding a form date cutoff that I discovered in the LBD DED on the NACC site and removing comments about unused code.

In order to use this program, make sure the "-ftld" and either "-ivp" or "-fvp" flags are specified when using the regular redcap2nacc converter. 
The output will look like a normal fixed-width formatted file ready to submit to NACC. I have checked that the output data lines up with the correct columns, so it is behaving as expected.
If you pass in a form with invalid data (a form with a date that's before the form was published, a field with data that should be blank, a form that's somehow missing a field or a form from another module, etc) then redcap2nacc will print an error message and skip to the next ptid in the csv file.

I have tested this program with sample surveys I made on REDCap, with both a normal ptid and a ptid with some deliberate issues (numbers outside of the allowed range, invalid dates, fields with data that should have been blank). These issues were each found by redcap2nacc and skipped, while the ptid with normal data was processed each time. The output has the correct data in the correct columns. I have also made sure that redcap2nacc still runs as normal for the other modules.

There was some confusion about the blanking rules in this module- many fields have the rule "Blank if question not answered," in which case I told the blanks file to act as if there were no blanking rules. There was one field 'FTDCPC2F' that had blanking rules that depended on questions that came after it (which, in turn, had blanking rules that depended on _this_ field). For now, I have also passed this one through the "no blanking rules" function.

* [+] I matched the style of the existing code.
* [+] I added and updated any relevant documentation (inline, `README`, `CHANGELOG`, and such).
* [+] I used Python's type hinting.
* [+] I ran the automated tests and ensured they **ALL** passed.
* [+] I ran the linter and ensured my changes have not introduced **ANY** warnings or errors. (No new warnings were introduced. I adjusted some of the old code's strings in accordance with the "line too long" warning, but there were some logic statements I couldn't break into multiple lines.)
* [+] I have made an effort to minimize code changes and have not included any cruft (preference files, *.pyc files, old comments, print-debugging, unused variables).
* [+] I have made an effort maintain a clear commit history (haven't merged other branches or rebased improperly).
* [+] I have written the code myself or have given credit where credit is due.

### CODE REVIEW

File: tools/generator_python3.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python
```
Comment (ufl-taeber) [inline tools/generator_python3.py]: Please remove the shebang lines.

File: tools/generator_python3.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+# coding=ascii
```
Comment (ufl-taeber) [inline tools/generator_python3.py]: Default encoding for Python 3 is now UTF-8. I'm pretty sure this is ignored or at the vary least confusing as your text editor is using UTF-8 as well.

Yes, I know that I had originally added these two lines back in 2015, but times have changed. These are no longer desired or required.

File: tools/generator_python3.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+###############################################################################
+# Copyright 2015-2016 University of Florida. All rights reserved.
```
Comment (ufl-taeber) [inline tools/generator_python3.py]: Please update any copyright to read 2015-2019 for any files you add or change. Do not change files that are unrelated.

File: tools/generator_python3.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+# Use of this source code is governed by the license found in the LICENSE file.
+###############################################################################
+
+import csv
+import os
+import re
+import sys
+# The output is too big for the command terminal, so I'm going to 
+# print everything to 'output.txt' located in the same directory.
+sys.stdout = open('output.txt', 'wt')
```
Comment (ufl-taeber) [inline tools/generator_python3.py]: What do you mean output is too big for the terminal?

Anyway, don't do this as it changes `stdout` for the entire program and breaks the documented usage and expected behavior.
```suggestion
```

File: tools/generator_python3.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+class DynamicObject(object):
```
Comment (ufl-taeber) [inline tools/generator_python3.py]: Is this a conversion of `tools/generator.py` for python3? I guess it was missed somehow in the `2to3` conversion? If so, this needs to replace the original file.

File: tools/generator_python3.py [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+    # Extension to the standard object class
+    # Mainly used to allow dynamic addition of members.
```
Comment (ufl-taeber) [inline tools/generator_python3.py]: This should be a docstring.

File: tools/generator_python3.py [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+    #     self. id = []
+    #     self.name = []
+    #     self.order = []
+    #     self.type = []
+    #     self.length = []
+    #     self.position = ()
+    #     self.inclusive_range = ()
+    #     self.allowable_codes = []
+    #     self.blanks = []
+
```
Comment (ufl-taeber) [inline tools/generator_python3.py]: This is cruft. Remove.

File: tools/generator_python3.py (hunk 4 above)
Comment (s-emerson) [reply tools/generator_python3.py]: i meant that the terminal's history cuts off the top portion of the generator's output in vscode, so i can't copy the full output after running the generator. i will get rid of the line.

File: tools/generator_python3.py (hunk 5 above)
Comment (s-emerson) [reply tools/generator_python3.py]: yes, it's a conversion. i will replace the original.
